Content.Server/Atmos/Monitor/Systems/AirAlarmSystem.cs

```diff
@@ -295,6 +296,31 @@ private void OnUpdateDeviceData(EntityUid uid, AirAlarmComponent component, AirA
         else
             UpdateUI(uid, component);
     }
+	
+	private void OnCopyDeviceData(EntityUid uid, AirAlarmComponent component, AirAlarmCopyDeviceDataMessage args)
+    {
+        if (AccessCheck(uid, args.Session.AttachedEntity, component))
+        {
+			switch (args.Data)
+			{
+				case GasVentPumpData ventData:
+					foreach (string addr in component.VentData.Keys)
+					{
+						SetData(uid, addr, args.Data);
+					}
+					break;
+				case GasVentScrubberData scrubberData:
+					foreach (string addr in component.ScrubberData.Keys)
+					{
+						SetData(uid, addr, args.Data);
+					}
+					break;
+			}
+			
+		}
+        else
+            UpdateUI(uid, component);
```

Comment:
format this better, i.e 
```suggestion
        if (!AccessCheck(uid, args.Session.AttachedEntity, component))
        {
           UpdateUI(uid, component);
        	return;       
        }

		switch (args.Data)
		{
			case GasVentPumpData ventData:
				foreach (string addr in component.VentData.Keys)
				{
					SetData(uid, addr, args.Data);
				}
				break;
				
			case GasVentScrubberData scrubberData:
				foreach (string addr in component.ScrubberData.Keys)
				{
					SetData(uid, addr, args.Data);
				}
				break;
		}
	}
```